component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -114,40 +254,61 @@ public boolean canHandle(HttpServletRequest request) {
+                || StringUtils.isNotEmpty(request.getParameter(CODE))
+                || StringUtils.isNotEmpty(request.getParameter(EMAIL_ADDRESS))
+                || StringUtils.isNotEmpty(request.getParameter(USER_NAME)));
     }
 
     @Override
-    public AuthenticatorFlowStatus process(HttpServletRequest request,
-                                           HttpServletResponse response,
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
                                            AuthenticationContext context)
             throws AuthenticationFailedException, LogoutFailedException {
         // if the logout request comes, then no need to go through and complete the flow.
         if (context.isLogoutRequest()) {
             return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
-        } else if (StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.EMAIL_ADDRESS))) {
+        } else if (StringUtils.isNotEmpty(request.getParameter(EMAIL_ADDRESS))) {
             // if the request comes with EMAIL ADDRESS, it will go through this flow.
             initiateAuthenticationRequest(request, response, context);
             return AuthenticatorFlowStatus.INCOMPLETE;
-        } else if (StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.CODE)) &&
-                StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.RESEND))) {
+        } else if (StringUtils.isEmpty(request.getParameter(CODE)) && StringUtils.isEmpty(request.getParameter(RESEND)))
+        {
             // if the request comes with code, it will go through this flow.
-            initiateAuthenticationRequest(request, response, context);
+            if (!isIdfInitiatedFromEmailOTP(context)) {
+                if (context.getLastAuthenticatedUser() == null) {
+                    context.setProperty(WITHOUT_AUTHENTICATED_USER, true);
+                    redirectUserToIdf(request, response, context);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+                }
+            }
+            if (context.getLastAuthenticatedUser() == null) {
+                org.wso2.carbon.user.core.common.User user = resolveUser(request, context);
+                setResolvedUserInContext(context, user);
+            }
+
+            try {
+                initiateAuthenticationRequest(request, response, context);
+            } catch (AuthenticationFailedException e) {
+                // The case where user has entered an invalid username
+                log.error(e);
+                Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
+                String queryParams = FrameworkUtils.getQueryStringWithFrameworkContextId(context.getQueryParams(),
+                        context.getCallerSessionKey(), context.getContextIdentifier());
+                String dummyEmail = "dummy@email.com";
```

Comment:
Why is there a hardcoded dymmy@email.com is defined here and passed to the `redirectToEmailOTPLoginPage` method?

Reply:
The `redirectToEmailOTPLoginPage` method is the method which redirects the user to the page where the user has to enter the OTP received via email. Since it is already present in the code, instead of creating a separate method I used the same one with a dummy email.

Reply:
Here is the case when a user is not resolved from the request and thrown an exception. So a valid email cannot be obtained to be input to the redirection method. That is the reason I used a dummy value. There is no other instance where a different value can come into this place, as the email.
A part of the email address can be created from the request itself, but still the rest will have to be a dummy value, because in a scenario where the requested user is not there, no email can be resolved in a logical manner.
Kindly give further suggestions on resolving this scenario.